rest_configuration.cpp

```diff
@@ -253,6 +257,94 @@ void DeRestPluginPrivate::initTimezone()
     daylightTimerFired();
 }
 
+/*! Init the network info. */
+void DeRestPluginPrivate::initNetworkInfo()
+{
+    bool ok = false;
+    bool retry = false;
+
+    QList<QNetworkInterface> ifaces = QNetworkInterface::allInterfaces();
+    QList<QNetworkInterface>::Iterator i = ifaces.begin();
+    QList<QNetworkInterface>::Iterator end = ifaces.end();
+
+    // optimistic approach chose the first available ethernet interface
+    for (; !ok && i != end; ++i)
+    {
+        if (i->name() == QLatin1String("tun0"))
+        {
+            continue;
+        }
+
+        retry = true;
+        if ((i->flags() & QNetworkInterface::IsUp) &&
+            (i->flags() & QNetworkInterface::IsRunning) &&
+            !(i->flags() & QNetworkInterface::IsLoopBack))
+        {
+            //DBG_Printf(DBG_INFO, "%s (%s)\n", qPrintable(i->name()), qPrintable(i->humanReadableName()));
+
+            QList<QNetworkAddressEntry> addresses = i->addressEntries();
+
+            if (ok || addresses.isEmpty())
+            {
+                continue;
+            }
+
+            QList<QNetworkAddressEntry>::Iterator a = addresses.begin();
+            QList<QNetworkAddressEntry>::Iterator aend = addresses.end();
+
+            for (; a != aend; ++a)
+            {
+                if (a->ip().protocol() != QAbstractSocket::IPv4Protocol)
+                {
+                    continue;
+                }
+
+                quint32 ipv4 = a->ip().toIPv4Address();
+                if ((ipv4 & 0xff000000UL) == 0x7f000000UL)
+                {
+                    // 127.x.x.x
+                    continue;
+                }
+
+                if ((ipv4 & 0x80000000UL) != 0x00000000UL && // class A 0xxx xxxx
+                    (ipv4 & 0xc0000000UL) != 0x80000000UL && // class B 10xx xxxx
+                    (ipv4 & 0xe0000000UL) != 0xc0000000UL)   // class C 110x xxxx
+                {
+                    // unsupported network
+                    continue;
+                }
+
+                QString mac = i->hardwareAddress().toLower();
+                gwMAC = mac;
+                gwBridgeId = mac.mid(0,2) + mac.mid(3,2) + mac.mid(6,2) + "fffe" + mac.mid(9,2) + mac.mid(12,2) + mac.mid(15,2);
+                if (gwConfig.contains("bridgeid") || gwConfig["bridgeid"] != gwBridgeId)

```

Comment:
! missing?

if (!gwConfig.contains() ...

Reply:
Thanks, good spot. I've amended the commit.